contracts/integrations/test/fuzz_tests/exchange_signature_validation_test.ts

```diff
@@ -0,0 +1,832 @@
+import { blockchainTests, constants, expect, signingUtils, transactionHashUtils } from '@0x/contracts-test-utils';
+import { orderHashUtils } from '@0x/order-utils';
+import { EIP712DomainWithDefaultSchema, Order, SignatureType, ZeroExTransaction } from '@0x/types';
+import { hexUtils, logUtils } from '@0x/utils';
+import * as ethUtil from 'ethereumjs-util';
+import * as _ from 'lodash';
+
+import { artifacts } from '../artifacts';
+import { Actor, ActorConfig } from '../framework/actors/base';
+import { AssertionResult } from '../framework/assertions/function_assertion';
+import { BlockchainBalanceStore } from '../framework/balances/blockchain_balance_store';
+import { DeploymentManager } from '../framework/deployment_manager';
+import { Simulation, SimulationEnvironment } from '../framework/simulation';
+import { Pseudorandom } from '../framework/utils/pseudorandom';
+import { TestSignatureValidationWalletContract } from '../wrappers';
+
+// tslint:disable: max-classes-per-file no-non-null-assertion no-unnecessary-type-assertion
+
+interface SignatureValidatorActorConfig extends ActorConfig {
+    walletContractAddress: string;
+    notWalletContractAddress: string;
+}
+
+class SignatureValidatorActor extends Actor {
+    protected readonly _accounts: { [address: string]: Buffer };
+    protected readonly _walletContractAddress: string;
+    protected readonly _notWalletContractAddress: string;
+
+    public constructor(config: SignatureValidatorActorConfig) {
+        super(config);
+        this.mixins.push('exchange/SignatureValidator');
+        this._accounts = _.zipObject(
+            this.deployment.accounts,
+            this.deployment.accounts.map((a, i) => constants.TESTRPC_PRIVATE_KEYS[i]),
+        );
+        this._walletContractAddress = config.walletContractAddress;
+        this._notWalletContractAddress = config.notWalletContractAddress;
+    }
+
+    protected async _presignHashAsync(signer: string, hash: string): Promise<void> {
+        await this.deployment.exchange.preSign(hash).awaitTransactionSuccessAsync({ from: signer });
+    }
+
+    protected async _approveValidatorAsync(signer: string, validator: string, approved: boolean = true): Promise<void> {
+        await this.deployment.exchange
+            .setSignatureValidatorApproval(validator, approved)
+            .awaitTransactionSuccessAsync({ from: signer });
+    }
```

Comment:
do these need to be wrapped in an actor at all? I think we can just have these as standalone generators